### PR TITLE
Fix building on ARM

### PR DESCRIPTION
**Pull request by dotdash**

### CODE REVIEW

File: src/librustc_llvm/lib.rs [hunk 1]
```diff
@@ -964,9 +964,10 @@ extern {
     pub fn LLVMAddFunctionAttrStringValue(Fn: ValueRef, index: c_uint,
                                           Name: *const c_char,
                                           Value: *const c_char);
+    pub fn LLVMRemoveFunctionAttributes(Fn: ValueRef, index: c_uint, attr: uint64_t);
     pub fn LLVMRemoveFunctionAttrString(Fn: ValueRef, index: c_uint, Name: *const c_char);
-    pub fn LLVMGetFunctionAttr(Fn: ValueRef) -> c_ulonglong;
-    pub fn LLVMRemoveFunctionAttr(Fn: ValueRef, val: c_ulonglong);
+    pub fn LLVMGetFunctionAttr(Fn: ValueRef) -> c_uint;
+    pub fn LLVMRemoveFunctionAttr(Fn: ValueRef, val: c_uint);
```
Comment (eddyb) [inline src/librustc_llvm/lib.rs]: Ouch, #17795 strikes again!

Comment (japaric) [inline src/librustc_llvm/lib.rs]: @alexcrichton would it be possible to use ctest to check the validity of these LLVM bindings?

Comment (alexcrichton) [inline src/librustc_llvm/lib.rs]: In theory yeah, but may be nontrivial to integrate

File: src/librustc_llvm/lib.rs [hunk 2]
```diff
@@ -964,9 +964,10 @@ extern {
     pub fn LLVMAddFunctionAttrStringValue(Fn: ValueRef, index: c_uint,
                                           Name: *const c_char,
                                           Value: *const c_char);
+    pub fn LLVMRemoveFunctionAttributes(Fn: ValueRef, index: c_uint, attr: uint64_t);
```
Comment (tamird) [inline src/librustc_llvm/lib.rs]: how come `attr` doens't need to be `c_ulonglong`?

Comment (dotdash) [inline src/librustc_llvm/lib.rs]: The C side uses `uint64_t` not `unsigned long long`.

**Pull request merged**